packages/tools/viewer-alpha/src/viewer.ts

```diff
@@ -891,6 +919,29 @@ export class Viewer implements IDisposable {
         return true;
     }
 
+    private _markSceneMutated() {
+        this._details.scene.incrementRenderId();
+        this._details.scene.executeWhenReady(() => {
+            this._sceneMutated = true;
+        });

```

Comment:
You should set the second parameter of `executeWhenReady` to `true`, to ensure that additional render targets (if any) are also ready. I think we should also make this change in the snapshot rendering helper, as we use `executeWhenReady` twice in this case.

Reply:
Would we expect additional render targets in the constrained case of the Viewer? Or would that require a user to do more advanced things, directly modifying the Babylon scene?

Reply:
There could be some for shadows or diffuse_transmission in gltf for instance.